src/components/transactions/TransactionRow.vue

```diff
@@ -355,31 +396,20 @@ export default {
       // Always retrieve transaction block hash as data.transaction.blockHash
       // may not be the final block hash in RSK
       const receipt = await originWeb3.eth.getTransactionReceipt(data.transaction.transactionHash)
-      const eventJsonInterface = BRIDGE_ABI.find(x => x.name === 'Cross' && x.type === 'event')
-      const eventSignature = originWeb3.eth.abi.encodeEventSignature(eventJsonInterface)
-      const event = receipt.logs.find(x => x.topics[0] === eventSignature)
-      event.topics.shift()
-      const decodedEvent = originWeb3.eth.abi.decodeLog(
-        eventJsonInterface.inputs,
-        event.data,
-        event.topics,
+      const { decodedEvent, event } = decodeCrossEvent(
+        originWeb3,
+        receipt,
+        this.globalState.currentTokenType,
       )
-      const gasPrice = await store.getGasPriceHex()
+      const tokenInstance = this.getTokenTypeInstance({ config: data.toNetwork })
 
       try {
-        const receiptResponse = await claim(
-          { config: data.toNetwork, web3: sharedState.web3 },
-          {
-            to: decodedEvent._to,
-            amount: decodedEvent._amount,
-            blockHash: event.blockHash,
-            transactionHash: event.transactionHash,
-            logIndex: event.logIndex,
-            txExplorerLink: data.txExplorerLinkTag,
-          },
-          { from: sharedState.accountAddress, gasPrice: gasPrice, gas: ESTIMATED_GAS_AVG },
+        // TODO: Generate instance to every kind of token and replace claim method
```

Comment:
if this still applies, I'd remove it and add a Github issue and/or a Jira ticket (tech debt)